cpp/src/arrow/util/decimal.h

```diff
@@ -40,43 +37,37 @@ namespace arrow {
 /// Semi-numerical Algorithms section 4.3.1.
 ///
 /// Adapted from the Apache ORC C++ implementation
-class ARROW_EXPORT Decimal128 {
+///
+/// The implementation is split into two parts :
+///
+/// 1. DecimalBasic128
+///    - can be safely compiled to IR without references to libstdc++.
+/// 2. Decimal128
+///    - has additional functionality on top of DecimalBasic128 to deal with
+///      strings and streams.
+class ARROW_EXPORT Decimal128 : public DecimalBasic128 {
  public:
   /// \brief Create an Decimal128 from the two's complement representation.
   constexpr Decimal128(int64_t high, uint64_t low) noexcept
-      : low_bits_(low), high_bits_(high) {}
+      : DecimalBasic128(high, low) {}
```

Comment:
fixed

Reply:
Looks like I spoke optimistically, sorry :-/ Compilation is failing on CI because of failing to resolve no-argument constructor calls. See e.g. https://travis-ci.org/apache/arrow/jobs/483403588#L1645

@bkietz Would you know what the proper fix is?

Reply:
it worked fine in CI on linux with clang 6.0, and on Mac with Xcode 8.3. so, seems to be an issue with some compilers.

I'm planning to add only the empty constructor back.

 constexpr Decimal128() noexcept : BasicDecimal128() {}